components/data-table/index.jsx

```diff
@@ -359,9 +425,141 @@ class DataTable extends React.Component {
+				[KEYS.ESCAPE]: { callback: (evt) => this.handleKeyDownEscape(evt) },
+			},
+		});
+	}
+
+	handleKeyDownUp() {
+		const newRowIndex = Math.max(this.state.activeCell.rowIndex - 1, 0);
+		const activeElement = this.getFirstInteractiveElement(
+			newRowIndex,
+			this.state.activeCell.columnIndex
+		);
+		if (newRowIndex !== this.state.activeCell.newRowIndex) {
+			this.setState((prevState) => ({
+				activeCell: {
+					rowIndex: newRowIndex,
+					columnIndex: prevState.activeCell.columnIndex,
+				},
+				activeElement,
+			}));
+		}
+	}
+
+	handleKeyDownDown() {
+		const newRowIndex = Math.min(
+			this.state.activeCell.rowIndex + 1,
+			this.props.items.length
+		);
+		const activeElement = this.getFirstInteractiveElement(
+			newRowIndex,
+			this.state.activeCell.columnIndex
+		);
+		if (newRowIndex !== this.state.activeCell.newRowIndex) {
+			this.setState((prevState) => ({
+				activeCell: {
+					rowIndex: newRowIndex,
+					columnIndex: prevState.activeCell.columnIndex,
+				},
+				activeElement,
+			}));
+		}
+	}
+
+	handleKeyDownLeft() {
+		const newColumnIndex = Math.max(this.state.activeCell.columnIndex - 1, 0);
+		const activeElement = this.getFirstInteractiveElement(
+			this.state.activeCell.rowIndex,
+			newColumnIndex
+		);
+		if (newColumnIndex !== this.state.activeCell.columnIndex) {
+			this.setState((prevState) => ({
+				activeCell: {
+					rowIndex: prevState.activeCell.rowIndex,
+					columnIndex: newColumnIndex,
+				},
+				activeElement,
+			}));
+		}
+	}
+
+	handleKeyDownRight() {
+		const newColumnIndex = Math.min(
+			this.state.activeCell.columnIndex + 1,
+			this.props.children.length
+		);
+		const activeElement = this.getFirstInteractiveElement(
+			this.state.activeCell.rowIndex,
+			newColumnIndex
+		);
+		if (newColumnIndex !== this.state.activeCell.columnIndex) {
+			this.setState((prevState) => ({
+				activeCell: {
+					rowIndex: prevState.activeCell.rowIndex,
+					columnIndex: newColumnIndex,
+				},
+				activeElement,
+			}));
+		}
+	}
+
+	handleKeyDownEnter() {
+		if (this.state.mode === Mode.NAVIGATION) {
+			const { rowIndex, columnIndex } = this.state.activeCell;
+			let activeElement = null;
+			if (this.interactiveElements[rowIndex][columnIndex]) {
+				[activeElement] = this.interactiveElements[rowIndex][columnIndex];
+			}
+			this.setState({
+				mode: Mode.ACTIONABLE,
+				activeElement,
+			});
+		}
+	}
+
+	handleKeyDownEscape() {
+		if (this.state.mode === Mode.ACTIONABLE) {
+			this.setState({
+				mode: Mode.NAVIGATION,
+				activeElement: null,
+			});
+		}
+	}
+
+	registerInteractiveElement(rowIndex, columnIndex, elementId) {
+		if (!this.interactiveElements[rowIndex]) {
+			this.interactiveElements[rowIndex] = {};
+		}
+		if (!this.interactiveElements[rowIndex][columnIndex]) {
+			this.interactiveElements[rowIndex][columnIndex] = [];
+		}
+		this.interactiveElements[rowIndex][columnIndex].push(elementId);
+	}
+
 	// ### Render
 	render() {
 		const ariaProps = {};
```

Comment:
We can reset interactiveElements at render time to avoid it getting larger on every render (potential memory leak)

```suggestion
		this.interactiveElements = {};
		const ariaProps = {};
```